Atomic/syscontainers.py

```diff
@@ -286,6 +286,52 @@ def _gather_installed_files_info (checkout, name):
 
         return info, rpm_installed, installed_files_checksum
 
+    def _get_remote_location(self, remote_input):
+        """
+        Parse the remote input and return actual remote path
+
+        :param remote_input: input path from user
+        :returns: the parsed remote location
+        """
+
+        if not remote_input:
+            return None
+
+        remote_path = os.path.realpath(remote_input)
+        if not os.path.exists(remote_path):
+            raise ValueError("The container's rootfs is set to remote, but the remote rootfs does not exist")
+
+        # Here we know remote path does exist on the fs
+        remote_rootfs = os.path.sep.join([remote_path, "rootfs"])
+        if os.path.exists(remote_rootfs):
+            util.write_out("The remote rootfs for this container is set to be {}".format(remote_rootfs))
+        elif os.path.exists(os.path.sep.join([remote_path, "usr"])):  # Assume that the user directly gave the location of the rootfs
+            remote_path = os.path.dirname(remote_path)  # Use the parent directory as the "container location"
+        else:
+            raise ValueError("--remote was specified but the given location does not contain a rootfs")
+        return remote_path
+
+    @staticmethod
+    def _rewrite_rootfs(destination, remote_rootfs):
+        """
+        When remote rootfs is specified, we rewrite the ['root']['path']
+
+        :param destination: the destination of the container
+        :param remote_rootfs: remote rootfs location
+        """
+        destination_path = os.path.join(destination, "config.json")
```

Comment:
Note: destination path is "generated" here instead of being a parameter passed in. This is because, in the future refactoring I plan to remove "destination_path" from local variable, to reach "separation of logic" and reduce the number of parameters/local variables. I am curious for your opinions, WDYT?  (I am ok with passing in as a parameter and save as a local variable as well)

By removing destination path, we can also save/remove two duplicated pieces, 
1: https://github.com/projectatomic/atomic/blob/master/Atomic/syscontainers.py#L380
2: https://github.com/projectatomic/atomic/blob/master/Atomic/syscontainers.py#L914